### PR TITLE
Update docs to use the new `teal_data` instead of `TealData`

### LINKED ISSUE #591: Replace TealData with teal_data in the docs
Will be needed soon to enable removal of the old classes in `teal_data`.

### PR DESCRIPTION
**Pull request by vedhav**
Closes #591

### CODE REVIEW

File: R/tm_a_pca.R [hunk 1]
```diff
@@ -24,24 +24,23 @@
...
-#'   ),
+#'   data = data,
 #'   modules = teal::modules(
 #'     teal.modules.general::tm_a_pca(
 #'       "PCA",
 #'       dat = teal.transform::data_extract_spec(
 #'         dataname = "ADSL",
 #'         select = teal.transform::select_spec(
-#'           choices = teal.transform::variable_choices(data = ADSL, c("BMRKR1", "AGE", "EOSDY")),
+#'           choices = teal.transform::variable_choices(data = data@env$ADSL, c("BMRKR1", "AGE", "EOSDY")),
```
Comment (gogonzo) [inline R/tm_a_pca.R:43]:
```suggestion
#'           choices = teal.transform::variable_choices(data = data[["ADSL"]], c("BMRKR1", "AGE", "EOSDY")),
```

File: R/tm_a_regression.R [hunk 2]
```diff
@@ -34,13 +34,17 @@
...
 #'
-#' ADSL <- teal.modules.general::rADSL
+#' data <- teal_data()
+#' data <- within(data, {
+#'   library(nestcolor)
+#'   ADSL <- teal.modules.general::rADSL
+#' })
+#' datanames <- c("ADSL")
+#' datanames(data) <- datanames
+#' data@join_keys <- cdisc_join_keys(!!!datanames)
```
Comment (gogonzo) [inline R/tm_a_regression.R:44]: I see the point now of waiting for change of the join_keys. Let's continue, maybe we will be able to merge them in the same time.

**Review by gogonzo — COMMENTED** [review]
please change `data@env$ADSL` -> `data[["ADSL"]]` everywhere. `qenv` has method `[[` and `get_data` to pull objects from environment.

File: R/tm_a_pca.R (hunk 1 above)
Comment (vedhav) [reply R/tm_a_pca.R:43]: Thanks! I didn't realize that we had this method. I'll change them everywhere.

**Comment by m7pr** [comment]
@vedhav should you update NEWS?

**Comment by m7pr** [comment]
should we bump teal.data version in description?

**Comment by gogonzo** [comment]
> should we bump teal.data version in description?

yes

**Comment by vedhav** [comment]
Closing in favor of #596

**Pull request closed without merging**